examples/playground/components/lazy/index.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { memo } from 'react';
+import { lazy, LazySuspense } from 'react-loosely-lazy';
+
+import { controlLoad } from '../../utils';
+import { Result, Progress } from '../result';
+
+const Async: any = {};
```

Comment:
You should be able to achieve the same goal without needing to have as many side effects or modifying this `Async` variable. Something like:


```
export const buildLazyComponents = {
  server: () => {
    const LazyWithoutSSR = lazy(() => import('./without-ssr'), {
      moduleId: './lazy/without-ssr',
      ssr: false,
    });

    return () => <LazyComponents><LazyWithoutSSR /></LazyComponents>
  },
  client: () => {
    const LazyWithoutSSR = lazy(
      () => import('./without-ssr').then(controlLoad),
      { moduleId: './lazy/without-ssr', ssr: false }
    );

    return () => <LazyComponents><LazyWithoutSSR /></LazyComponents>
  },
};
```

You'd need to modify the `buildServerComponents` and `buildClientComponents` to handle a return instead as well, but it should still be fairly encapsulated.

For other examples where multiple components are used, you can just make the `LazyComponents` take in additional component props as opposed to children.